ROX-14221: upgrade procedures for cloud (#78259)
openshift/openshift-docs: cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc

:_mod-docs-content-type: ASSEMBLY
[id="upgrade-cloudsvc-operator"]
= Upgrading secured clusters in RHACS Cloud Service by using the Operator
include::modules/common-attributes.adoc[]
:context: upgrade-cloudsvc-operator

toc::[]

[role="_abstract"]

Red{nbsp}Hat provides regular service updates for the components that it manages, including Central services. These service updates include upgrades to new versions of {product-title-managed}.

You are responsible for regularly upgrading the version of {product-title-short} on your secured clusters to ensure compatibility with {product-title-managed-short}.

include::modules/prepare-operator-upgrades.adoc[leveloffset=+1]
include::modules/change-collection-method.adoc[leveloffset=+2]

[role="_additional-resources"]
.Additional resources
* link:https://docs.openshift.com/container-platform/latest/operators/admin/olm-upgrading-operators.html[Updating installed Operators]

> //does this apply to Cloud Service? Or only updating Central components?

Reviewer: Good question. I doubt that there are ACS CS customers that have secured clusters on the 3.7x versions (possibly some remaining from the "service preview" in late 2022), so we could skip this part about changing channels.

include::modules/operator-upgrade-change-subscription-channel.adoc[leveloffset=+1]

[id="rollback-operator-upgrade-cloud"]

> == Rolling back an Operator upgrade for secured clusters

Reviewer: I wonder if we should have this section.

We try to not roll back Central Services in ACS CS. We have canary deployments and we wouldn't deploy a new version to customers if it's problematic. Which should mean that customers shouldn't have to roll back their secured clusters either.

OTOH, there could be the possibility that there's a problem with the secured cluster specifically and has to be rolled back, so we could also keep this section.

:_mod-docs-content-type: ASSEMBLY
[id="upgrade-cloudsvc-operator"]
= Upgrading secured clusters in RHACS Cloud Service by using the Operator
include::modules/common-attributes.adoc[]
:context: upgrade-cloudsvc-operator

toc::[]

[role="_abstract"]

Red{nbsp}Hat provides regular service updates for the components that it manages, including Central services. These service updates include upgrades to new versions of {product-title-managed}.

> You are responsible for regularly upgrading the version of {product-title-short} on your secured clusters to ensure compatibility with {product-title-managed-short}.

Reviewer 2: ```suggestion
You must regularly upgrade the version of {product-title-short} on your secured clusters to ensure compatibility with {product-title-managed-short}.
```

include::modules/prepare-operator-upgrades.adoc[leveloffset=+1]
include::modules/change-collection-method.adoc[leveloffset=+2]

[role="_additional-resources"]
.Additional resources
* link:https://docs.openshift.com/container-platform/latest/operators/admin/olm-upgrading-operators.html[Updating installed Operators]

[id="rollback-operator-upgrade-cloud"]
== Rolling back an Operator upgrade for secured clusters

> To roll back an Operator upgrade, you must perform the steps described in one of the following sections. You can roll back an Operator upgrade by using the CLI or the {ocp} web console.

Reviewer 2: ```suggestion
To roll back an Operator upgrade, you can use either the CLI or the {ocp} web console.
```

[NOTE]
====

> Rolling back Operator upgrades on secured clusters is needed only in rare cases, for example, if there is an issue with the secured cluster.

Reviewer 2: ```suggestion
On secured clusters, rolling back Operator upgrades is needed only in rare cases, for example, if an issue exists with the secured cluster.
```
